src/org/sosy_lab/common/collect/OrderStatisticMapTestSuite.java

```diff
@@ -0,0 +1,369 @@
+import com.google.errorprone.annotations.Var;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NavigableMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+public abstract class OrderStatisticMapTestSuite {
+
+  @SuppressWarnings("unchecked")
+  private static final ImmutableList<Entry<String, String>> ELEMS =
+      ImmutableList.<Entry<String, String>>builder()
+          .add(Maps.immutableEntry("aaa", "Vzza"))
+          .add(Maps.immutableEntry("hha", "Vppa"))
+          .add(Maps.immutableEntry("ppa", "Vhha"))
+          .add(Maps.immutableEntry("zza", "Vaaa"))
+          .build();
+
+  @SuppressWarnings("unchecked")
+  private static final ImmutableList<Entry<String, String>> ELEMS_ABOVE =
+      ImmutableList.<Entry<String, String>>builder()
+          .add(Maps.immutableEntry("aab", "Vzzb"))
+          .add(Maps.immutableEntry("hhb", "Vppb"))
+          .add(Maps.immutableEntry("ppb", "Vhhb"))
+          .add(Maps.immutableEntry("zzb", "Vaab"))
+          .build();
+
+  @SuppressWarnings("unchecked")
+  private static final ImmutableList<Entry<String, String>> ELEMS_BELOW =
+      ImmutableList.<Entry<String, String>>builder()
+          .add(Maps.immutableEntry("aa", "Vzz"))
+          .add(Maps.immutableEntry("hh", "Vpp"))
+          .add(Maps.immutableEntry("pp", "Vhh"))
+          .add(Maps.immutableEntry("zz", "Vaa"))
+          .build();
+
+  public abstract static class OrderStatisticMapFactory extends TestStringSortedMapGenerator {
+
+    @Override
+    protected abstract OrderStatisticMap<String, String> create(Entry<String, String>[] pEntries);
+
+    protected abstract OrderStatisticMap<String, String> create(
+        List<Entry<String, String>> pEntries);
+  }
+
+  private OrderStatisticMapFactory factory;
+
+  public OrderStatisticMapTestSuite(OrderStatisticMapFactory pFactory) {
+    factory = pFactory;
+  }
+
+  @SuppressWarnings("unchecked")
+  private OrderStatisticMap<String, String> createMap() {
+    return factory.create(Collections.emptyList());
+  }
+
+  @SuppressWarnings("unchecked")
+  private OrderStatisticMap<String, String> createMap(List<Entry<String, String>> pEntries) {
+    return factory.create(pEntries);
+  }
+
+  private static <K, V> boolean containsEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    return pMap.containsKey(pEntry.getKey()) && pEntry.getValue().equals(pMap.get(pEntry.getKey()));
+  }
+
+  private static <K, V> void putEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    pMap.put(pEntry.getKey(), pEntry.getValue());
+  }
+
+  private static <K, V> void removeEntry(Map<K, V> pMap, Entry<K, V> pEntry) {
+    Assert.assertEquals(pMap.get(pEntry.getKey()), pMap.remove(pEntry.getKey()));
+  }
+
+  @Test
+  public void testEquals() {
+    OrderStatisticMap<String, String> l1 = createMap();
+    @Var OrderStatisticMap<String, String> l2 = createMap();
+
+    // Check that the map sorts its elements
```

Comment:
`equals` of maps does not depend on ordering. I recommend `EqualsTester` for testing `equals`, because one should also check for example that such a map is equal to a `HashMap` with the same contents etc. But tests that the `NavigableMapTestSuiteBuilder` already does of course need not be written manually.